### ISSUE TITLE
Unnecessary use of aria-label

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15337]
Unnecessary use of aria-label

-   **Severity:** Low
-   **Affected Populations:**
    -   Blind
    -   Low-Vision
    -   Cognitively Impaired
-   **Platform(s):**
    -   Windows - Screen Reader
    -   Windows - Dragon
    -   Mac - VoiceOver
    -   Android - TalkBack
    -   iOS - VoiceOver
-   **Components affected:**
    -   Add Block

#### Issue description

The content inside the dropdown/popup of the "Add Block" button uses
redundant `aria-label` attributes on the block type buttons. This text
is also already visible even in the mobile view, and the SVGs are
correctly hidden from assistive technology, therefore the `aria-label`
text is not needed.

It appears the normally-unnecessary `role=list` on the unordered list
is essential to making Firefox expose the list semantics.

While redundant labels do not affect end-users, unnecessary
`aria-label` attributes create the possibility of mismatches between
visible labels and accessible names when content is updated.

##### Issue Code

```
    <button class="..." aria-label="Paragraph">


        <span class="...">


            <span class="...">


                <svg aria-hidden="true"...></svg>


            </span>


        </span>


        <span class="editor-block-types-list__item-title">Paragraph</span></button>
```

#### Remediation Guidance

Remove the `aria-label` from the button and allow the visible text
inside to label the button.

This prevents future potential mismatches between an invisible and a
visible name.

##### Recommended Code

```
    <button class="...">


        <span class="...">


            <span class="...">


                <svg aria-hidden="true"...></svg>


            </span>


        </span>


        <span class="editor-block-types-list__item-title">Paragraph</span></button>
```

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-18 in Tenon's report

**Comment by afercia** [comment]
> It appears the normally-unnecessary role=list on the unordered list
is essential to making Firefox expose the list semantics.

Yep, this is used in a few places because Safari doesn't expose _styled_ lists as lists. Not sure about Firefox.

**Comment by melchoyce** [comment]
Screenshot from full report:

![image](https://user-images.githubusercontent.com/2846578/57185654-ab49f500-6e84-11e9-9847-c1f2cc6a6f95.png)

**Comment by afercia** [comment]
Relevant current code:
https://github.com/WordPress/gutenberg/blob/4adbe72f846eff921630f101456d98d484df9fce/packages/block-editor/src/components/inserter-list-item/index.js#L30-L64

The `aria-label` and the visible text use the same variable. Originally introduced in #5828 to fix an issue specific to Internet Explorer 11 and JAWS, see #5827. Should be tested again to make sure it's still relevant. Although there have been many changes since then, they mostly apply to code reorganization and moving files. Functionally, not so much has changed and I guess this fix is still necessary.
